ui/src/pages/AllStories.tsx

```diff
@@ -0,0 +1,288 @@
+          [classes.noClick]: onRowClick == null,
+        })}
+        variant="body"
+        style={{ height: rowHeight }}
+        align={
+          (columnIndex != null && columns[columnIndex].numeric) || false
+            ? "right"
+            : "left"
+        }
+      >
+        {columns[columnIndex].toggle && (
+          <Switch disabled inputProps={{ "aria-label": "disabled checkbox" }} />
+        )}
+        {cellData}
+      </TableCell>
+    );
+  };
+
+  headerRenderer = ({
+    label,
+    columnIndex,
+  }: TableHeaderProps & { columnIndex: number }) => {
+    const { headerHeight, columns, classes } = this.props;
+
+    return (
+      <TableCell
+        component="div"
+        className={clsx(
+          classes.tableCell,
+          classes.flexContainer,
+          classes.noClick
+        )}
+        variant="head"
+        style={{ height: headerHeight }}
+        align={columns[columnIndex].numeric || false ? "right" : "left"}
+      >
+        <span>{label}</span>
+      </TableCell>
+    );
+  };
+
+  render() {
+    const {
+      classes,
+      columns,
+      rowHeight,
+      headerHeight,
+      ...tableProps
+    } = this.props;
+    return (
+      <AutoSizer>
+        {({ height, width }) => (
+          <Table
+            height={height}
+            width={width}
+            rowHeight={rowHeight}
+            gridStyle={{
+              direction: "inherit",
+            }}
+            headerHeight={headerHeight}
+            className={classes.table}
+            {...tableProps}
+            rowClassName={this.getRowClassName}
+          >
+            {columns.map(({ dataKey, ...other }, index) => {
+              return (
+                <Column
+                  key={dataKey}
+                  headerRenderer={(headerProps) =>
+                    this.headerRenderer({
+                      ...headerProps,
+                      columnIndex: index,
+                    })
+                  }
+                  className={classes.flexContainer}
+                  cellRenderer={this.cellRenderer}
+                  dataKey={dataKey}
+                  {...other}
+                />
+              );
+            })}
+          </Table>
+        )}
+      </AutoSizer>
+    );
+  }
+}
+
+const VirtualizedTable = withStyles(styles)(MuiVirtualizedTable);
+
+interface TableData {
+  id: number;
+  title: string;
+  city: string;
+  year: string;
+  author: string;
+  country: string;
+  is_visible: boolean;
+}
+
+function createData(
+  id: number,
+  title: string,
+  city: string,
+  year: string,
+  author: string,
+  country: string,
+  is_visible: boolean
+): TableData {
+  return { id, title, city, year, author, country, is_visible };
+}
+
+export const AllStories: React.FC = () => {
+  const { data: allStories, error } = useSWR<Story[]>("/api/stories");
+
+  if (allStories) {
+    const rows: TableData[] = [];
+    for (let i = 0; i < 33; i += 1) {
```

Comment:
Is the 33 here in the for statement because we currently have 33 stories? Can we put a variable instead (ie. allStories.length())?